### ISSUE TITLE
Missing Documentation for compiling binaries in c++

### ISSUE THREAD

**Issue by Joel-De** [tensorflow/tensorflow#45897]
Not sure if this is the right place to post this, but anyways, does anyone have a link for documentation on building TensorFlow for C++ to be used with VS 2019? I've looked everywhere and can't seem to find any up to date information on how to do so. The build from source page on TensorFlow's website seemed to look like it was just for building a pip installer for python. 

I followed along with 

https://medium.com/vitrox-publication/deep-learning-frameworks-tensorflow-build-from-source-on-windows-python-c-cpu-gpu-d3aa4d0772d8

but after linking the header files and .lib & .dll I got spewed a bunch of missing dependencies/syntax errors, so I'm guessing that that the bugs mentioned in the article didnt follow through to v2.4.0 (which is what I compiled it for)

If anyone could point to up to date documentation, or a pre-compiled library that works with c++ VS19 I'd appreciate it.

**Comment by amahendrakar** [comment]
@Joel-De,
Could you please take a look at this [TensorFlow C API](https://www.tensorflow.org/install/lang_c) installation guide and let us know if it helps. Thanks!

**Comment by Joel-De** [comment]
@amahendrakar 

I've looked through what you've provided, if I'm not mistaken that appears to be for the c api, I've tried using it for cpp but there are compile errors,

Error	LNK2019	unresolved external symbol __imp__TF_Version referenced in function _main
Error	LNK1120	1 unresolved externals	

I believe this has to do with improperly converting the c based api to one that c++ can work with (from my searches online it reffers to using the 'extern "C" ' keyword). Is there a way to use the c api files with c++? Or do I need to build those myself, and if so how would I do that?

As a note, my goal is to use c++ to perform inferences for a pretrained model built using the python api.

Thanks!
